app/models/concerns/foreman_remote_execution/smart_proxy_extensions.rb

```diff
@@ -0,0 +1,25 @@
+module ForemanRemoteExecution
+  module SmartProxyExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      alias_method_chain :refresh, :remote_execution
+    end
+
+    def pubkey
+      self[:pubkey] || update_pubkey
+    end
+
+    def update_pubkey
+      return unless has_feature?('Ssh')
+      key = ::ProxyAPI::RemoteExecutionSSH.new(:url => url).pubkey
+      self.update_attribute(:pubkey, key) if key
+      key
+    end
+
+    def refresh_with_remote_execution
+      update_pubkey
```

Comment:
in the case the proxy has the feature turned on, but foreman did not refresh yet, does it make sense to swap the order of methods?